sim.c

```diff
@@ -1,6 +1,6 @@
+#include "sim.h"
 #include "gbuffer.h"
 #include "mark.h"
-#include "sim.h"
```

Comment:
These headers should not be reordered like this -- clang-foramt is supposed to keep the .h for a .c at the top, then alphabetically sort the other ones below it.